cscs-checks/microbenchmarks/gpu/memory_bandwidth/src/types.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef __INCLUDED_TYPES__
+#define __INCLUDED_TYPES__
+
+#include "Xdevice/runtime.hpp"
+
+class RAIIData
+{
+public:
+  RAIIData() {};
+  virtual ~RAIIData() {};
+};
+
+// Class managing host data.
+class HostData : public RAIIData
```

Comment:
Do you really need to inherit here? Your class is RAII by its behaviour.

Reply:
Also instead of using your own RAII pointer management, why not use `std::unique_ptr`?

Reply:
Yeah, forgot to clean that up. The reason why I've gone for the RAII class instead of the `std::unique_ptr` is to sneak the memset right after the allocation.